### PR TITLE
fix InvertedIndex::Train when all input vectors are empty

### PR DESCRIPTION
**Pull request by zhengbuqian**
issue: https://github.com/zilliztech/knowhere/issues/774
/kind bug

### CODE REVIEW

File: src/index/sparse/sparse_inverted_index.h [hunk 1]
```diff
@@ -200,6 +200,9 @@ class InvertedIndex : public BaseInvertedIndex<T> {
         for (size_t i = 0; i < rows; ++i) {
             amount += data[i].size();
         }
+        if (amount == 0) {
+            return Status::success;
```
Comment (alexanderguzhva) [inline src/index/sparse/sparse_inverted_index.h:204]: this early exit does not set rhe following values (lines 222-223)
```
        value_threshold_ = *pos;
        drop_during_build_ = true;
```
is it expected?

Comment (zhengbuqian) [reply src/index/sparse/sparse_inverted_index.h:204]: yes, they have default values of 0 and false.

**Review by liliu-z — APPROVED** [review]
lgtm

**Pull request merged**